### PR TITLE
feat(radio-button)!: add icon to radio button

### PR DESCRIPTION
**Pull request by djcouto**
### Breaking Change
- `label` property is now **optional** and `value` is **required**

### Screenshot
Screenshot 2022-11-19 at 11 27 22

### CODE REVIEW

**Review by nelsonmestevao — APPROVED** [review]

**Comment by miguelpinto98** [comment]
Hmmm... Looking for the objective of this component, isn't possible to have icon & text at the same same? We are adding a `type` that enforce one of it, but seems possible to have both working, or independently without breaking changes. What you think about this @djcouto ?

File: packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx [hunk 1]
```diff
@@ -5,11 +5,11 @@ describe('Radio Button (unit tests)', () => {
 	let page: SpecPage;
 	let component: KvRadioButton;
 
-	describe('when the component loads with default props', () => {
+	describe('when the component loads with type text', () => {
 		beforeEach(async () => {
 			page = await newSpecPage({
 				components: [KvRadioButton],
-				html: '<kv-radio-button label="Option 1"></kv-radio-button>'
+				html: '<kv-radio-button type="text" label="Option 1" value="opt1"></kv-radio-button>'
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx:12]: remove `type="text"`

File: packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx [hunk 2]
```diff
@@ -19,11 +19,11 @@ describe('Radio Button (unit tests)', () => {
 		});
 	});
 
-	describe('when the component loads with disabled prop', () => {
+	describe('when the component loads with type icon', () => {
 		beforeEach(async () => {
 			page = await newSpecPage({
 				components: [KvRadioButton],
-				html: '<kv-radio-button label="Option 1" disabled></kv-radio-button>'
+				html: '<kv-radio-button type="icon" icon="kv-add" value="opt1"></kv-radio-button>'
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx:26]: remove `type="icon"`

File: packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx [hunk 3]
```diff
@@ -33,11 +33,11 @@ describe('Radio Button (unit tests)', () => {
 		});
 	});
 
-	describe('when the component loads with checked prop', () => {
+	describe('when the component loads with disabled prop', () => {
 		beforeEach(async () => {
 			page = await newSpecPage({
 				components: [KvRadioButton],
-				html: '<kv-radio-button label="Option 1" checked></kv-radio-button>'
+				html: '<kv-radio-button type="text" label="Option 1" value="opt1" disabled></kv-radio-button>'
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx:40]: remove `type="text"`

File: packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx [hunk 4]
```diff
@@ -47,11 +47,11 @@ describe('Radio Button (unit tests)', () => {
 		});
 	});
 
-	describe('when the component loads with value prop', () => {
+	describe('when the component loads with checked prop', () => {
 		beforeEach(async () => {
 			page = await newSpecPage({
 				components: [KvRadioButton],
-				html: '<kv-radio-button label="Option 1" value="opt1"></kv-radio-button>'
+				html: '<kv-radio-button type="text" label="Option 1" value="opt1" checked></kv-radio-button>'
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx:54]: remove `type="text"`

File: packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx [hunk 5]
```diff
@@ -65,7 +65,7 @@ describe('Radio Button (unit tests)', () => {
 		beforeEach(async () => {
 			page = await newSpecPage({
 				components: [KvRadioButton],
-				html: '<kv-radio-button label="Option 1" prevent-default="true"></kv-radio-button>'
+				html: '<kv-radio-button type="text" label="Option 1" value="opt1" prevent-default="true"></kv-radio-button>'
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx:68]: remove `type="text"`

Comment (ricardogoncalves89) [reply packages/ui-components/src/components/radio-button/test/radio-button.spec.tsx:68]: and add a case for `kv-radio-button` with both `icon` and `label` props

**Review by ricardogoncalves89 — APPROVED** [review]
LGTM 👍 just a few minor test fixes

File: packages/ui-components/src/components/icon/icon.scss [hunk 6]
```diff
@@ -11,6 +11,8 @@
 	--icon-height: 16px;
 	--icon-width: 16px;
 	--icon-rotation: 0deg;
+
+	display: flex;
```
Comment (pedrorebelocosta) [inline packages/ui-components/src/components/icon/icon.scss:15]: Is this change necessary?

**Review by pedrorebelocosta — APPROVED** [review]
Just a question, other than that LGTM :D

**Comment by miguelpinto98** [comment]
Please remove the breaking commit notation

**Review by miguelpinto98 — APPROVED** [review]
🚀  Only check the previous suggestions/comments from @ricardogoncalves89 and @pedrorebelocosta before merge

**Comment by djcouto** [comment]
@miguelpinto98
> Please remove the breaking commit notation

I can't, this PR introduces a breaking change. The `value` is now **required** and the `label` is now **optional**.
